Carrot-Assistant/app.py

```diff
@@ -191,52 +199,78 @@ async def generate_events(request: PipelineRequest) -> AsyncGenerator[str]:
             print(f"OMOP match found for {informal_name}: {omop_output}")
             output = {"event": "omop_output", "data": omop_output}
             yield json.dumps(output)
-            continue
-
         else:
-            print("No satisfactory OMOP results found for {informal_name}, using LLM...")
+            print(f"No satisfactory OMOP results found for {informal_name}")
+            output = {
+                "event": "omop_output",
+                "data": omop_output,
+                "message": f"No match found in OMOP database for {informal_name}.",
+            }
+            yield json.dumps(output)
+            no_match_names.append(informal_name)
 
+    
     # Use LLM to find the formal name and query OMOP for the LLM output
+    
+    if no_match_names and use_llm:
+        llm_outputs = assistant.run(
+            opt=opt, informal_names=informal_names, logger=logger
+        )
 
-    llm_outputs = assistant.run(opt=opt, informal_names=informal_names, logger=logger)
-    for llm_output in llm_outputs:
+        for llm_output in llm_outputs:
 
+            print(
+                "LLM output for", llm_output["informal_name"], ":", llm_output["reply"]
+            )
 
-        print("LLM output for", llm_output["informal_name"], ":", llm_output["reply"])
-        
-        print("Querying OMOP for LLM output:", llm_output["reply"])
+            print("Querying OMOP for LLM output:", llm_output["reply"])
 
-        output = {"event": "llm_output", "data": llm_output}
-        yield json.dumps(output)
+            output = {"event": "llm_output", "data": llm_output}
+            yield json.dumps(output)
 
-        # Simulate some delay before sending the next part
-        await asyncio.sleep(2)
+            # Simulate some delay before sending the next part
+            await asyncio.sleep(2)
 
-        omop_output = OMOP_match.run(
-            opt=opt, search_term=llm_output["reply"], logger=logger
-        )
+            omop_output = OMOP_match.run(
+                opt=opt, search_term=llm_output["reply"], logger=logger
+            )
 
-        print("OMOP output for", llm_output["reply"], ":", omop_output)
+            print("OMOP output for", llm_output["reply"], ":", omop_output)
+
+            output = {"event": "omop_output", "data": omop_output}
+            yield json.dumps(output)
 
-        output = {"event": "omop_output", "data": omop_output}
-        yield json.dumps(output)
 
 
 @app.post("/run")
-async def run_pipeline(request: PipelineRequest) -> EventSourceResponse:
+async def run_pipeline(request: Request) -> EventSourceResponse:
```

Comment:
Can you explain this change? Why doesn't the pipeline use a PipelineRequest?

Reply:
I believe before modification, we couldn't modify the API as the LLM uses the medication name to predict the formal name and queries the OMOP Database. By using "Request", I can send it to the OMOP Querying and if necessary (to use LLM), the "Request" will be convert into "PipelineRequest".